### PR TITLE
[script.domoticz.scenes] 0.0.10

### PR DESCRIPTION
**Pull request by Paul-VW**
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.domoticz.scenes/.gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+*.pyo
```
Comment (enen92) [inline script.domoticz.scenes/.gitignore:1]: This ha to be dropped from the submission

File: script.domoticz.scenes/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Domoticz Scenes
```
Comment (enen92) [inline script.domoticz.scenes/README.md:1]: License file is empty, please fix it

File: script.domoticz.scenes/addon.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+<addon id="script.domoticz.scenes" name="Domoticz Scenes" version="0.0.8" provider-name="Paul VW">
+  <requires>
+    <import addon="xbmc.python" version="3.0.0"/>
+     <import addon="script.module.requests"   version="1.0.0"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource" library="addon.py">
+    <provides>executable</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+    <summary lang="nl_NL">Start scenes in Domoticz vanuit Kodi</summary>
```
Comment (enen92) [inline script.domoticz.scenes/addon.xml:11]: en_GB versions is mandatory (default language)

File: script.domoticz.scenes/resources/lib/script.py [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+addonID = 'script.domoticz.scenes'
+addonVersion = '0.0.8'
+addonDate = "24 Maart 2021"
+
+__addon__ = xbmcaddon.Addon()
+__addonname__ = __addon__.getAddonInfo('name')
+__icon__ = __addon__.getAddonInfo('icon')
+
+LIB_DIR = xbmc.translatePath(os.path.join(xbmcaddon.Addon(id=addonID).getAddonInfo('path'), 'resources', 'lib'))
+sys.path.append(LIB_DIR)
```
Comment (enen92) [inline script.domoticz.scenes/resources/lib/script.py:21]: This should be xbmcvfs.translatePath

Comment (enen92) [inline script.domoticz.scenes/resources/lib/script.py:21]: Sys.path.append is a bad practice in Kodi add-ons, please use relative module imports

File: script.domoticz.scenes/resources/lib/script.py [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+                param[splitparams[0]] = splitparams[1]
+
+    return param
+
+
+params = get_params()
+
+
+def show_notification(msg):
+    xbmc.executebuiltin('Notification(' + __addonname__ + ',' + msg + ',5000,' + __icon__ + ')')
```
Comment (enen92) [inline script.domoticz.scenes/resources/lib/script.py:68]: xbmcgui.Dialog().notification

File: script.domoticz.scenes/resources/lib/script.py [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+    else:
+        base_url = "http://" + host + ":" + str(port)
+
+    if username is not None:
+        setcreds = 'username=' + username + '&password=' + password + '&'
+    else:
+        setcreds = ''
+
+    url = base_url + "/json.htm?" + setcreds + "type=command&param=switchlight&idx=" + str(idx) + "&switchcmd=Set%20Level&level=" + state
+    f = open(r"c:\tmp\debug.txt", "a")
```
Comment (enen92) [inline script.domoticz.scenes/resources/lib/script.py:236]: Use special://temp

File: script.domoticz.scenes/resources/lib/script.py [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+    f.write(url + "\n")
```
Comment (enen92) [inline script.domoticz.scenes/resources/lib/script.py:237]: Use python's with statement

File: script.domoticz.scenes/resources/lib/script.py [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+    f.close()
+    requests.get(url=url)
+
+
+def get_list(optionsDict):
+    optionsList = []
+    for line in optionsDict:
+        optionsList.append(line['Name'])
+    optionsList.append("------End of list------")
```
Comment (enen92) [inline script.domoticz.scenes/resources/lib/script.py:246]: If this is shown in the UI it has to be localized

File: script.domoticz.scenes/resources/lib/script.py [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+        base_url = "https://" + host + ":" + str(port)
+    else:
+        base_url = "http://" + host + ":" + str(port)
+
+    if username is not None:
+        setcreds = 'username=' + username + '&password=' + password + '&'
+    else:
+        setcreds = ''
+
+    url = base_url + "/json.htm?" + setcreds + "type=command&param=switchscene&idx=" + str(idx) + "&switchcmd=On"
```
Comment (enen92) [inline script.domoticz.scenes/resources/lib/script.py:199]: You have loads of code repetition. Wouldn't it make more sense to create a common method and only vary the argument passed to the URL?

**Review by enen92 — CHANGES_REQUESTED** [review]

File: script.domoticz.scenes/resources/lib/script.py (hunk 9 above)
Comment (Paul-VW) [reply script.domoticz.scenes/resources/lib/script.py:199]: Cleaned up

File: script.domoticz.scenes/resources/lib/script.py (hunk 8 above)
Comment (Paul-VW) [reply script.domoticz.scenes/resources/lib/script.py:246]: fixed

File: script.domoticz.scenes/resources/lib/script.py (hunk 7 above)
Comment (Paul-VW) [reply script.domoticz.scenes/resources/lib/script.py:237]: removed

File: script.domoticz.scenes/resources/lib/script.py (hunk 6 above)
Comment (Paul-VW) [reply script.domoticz.scenes/resources/lib/script.py:236]: removed

File: script.domoticz.scenes/resources/lib/script.py (hunk 5 above)
Comment (Paul-VW) [reply script.domoticz.scenes/resources/lib/script.py:68]: removed

File: script.domoticz.scenes/resources/lib/script.py (hunk 4 above)
Comment (Paul-VW) [reply script.domoticz.scenes/resources/lib/script.py:21]: removed

Comment (Paul-VW) [reply script.domoticz.scenes/resources/lib/script.py:21]: removed

File: script.domoticz.scenes/README.md (hunk 2 above)
Comment (Paul-VW) [reply script.domoticz.scenes/README.md:1]: fixed

File: script.domoticz.scenes/addon.xml (hunk 3 above)
Comment (Paul-VW) [reply script.domoticz.scenes/addon.xml:11]: added

File: script.domoticz.scenes/.gitignore (hunk 1 above)
Comment (Paul-VW) [reply script.domoticz.scenes/.gitignore:1]: fixed

File: script.domoticz.scenes/.gitignore [hunk 10]
```diff
@@ -0,0 +1,6 @@
+.project
```
Comment (enen92) [inline script.domoticz.scenes/.gitignore:1]: Drop from the submission...

**Review by enen92 — CHANGES_REQUESTED** [review]

File: script.domoticz.scenes/.gitignore (hunk 10 above)
Comment (Paul-VW) [reply script.domoticz.scenes/.gitignore]: dropped

File: script.domoticz.scenes/.gitignore [hunk 11]
```diff
@@ -0,0 +1,5 @@
+.pydevproject
```
Comment (enen92) [inline script.domoticz.scenes/.gitignore:1]: Still here...

**Review by enen92 — CHANGES_REQUESTED** [review]

File: script.domoticz.scenes/.gitignore (hunk 11 above)
Comment (Paul-VW) [reply script.domoticz.scenes/.gitignore]: should be fixed now

File: script.domoticz.scenes/.gitignore [hunk 12]
```diff
@@ -0,0 +1,4 @@
+pysrc
```
Comment (enen92) [inline script.domoticz.scenes/.gitignore:1]: ... remove the WHOLE gitignore file from the submission.

Comment (Paul-VW) [reply script.domoticz.scenes/.gitignore]: Removed

**Review by enen92 — APPROVED** [review]

**Pull request merged**